src/Microsoft.Azure.WebJobs.Host/Scale/ScaleManager.cs

```diff
@@ -0,0 +1,300 @@
+                        _logger.LogError(exc, $"Failed to query scale result for target scaler '{targetScaler.TargetScalerDescriptor.FunctionId}'.");
+                    }
+                }
+            }
+            return targetScaleVotes;
+        }
+
+        /// <summary>
+        /// Returns scale monitors and target scalers we want to use based on the configuration.
+        /// Scaler monitor will be ignored if a target scaler is defined in the same extensions assembly and TBS is enabled.
+        /// </summary>
+        internal static (List<IScaleMonitor>, List<ITargetScaler>) GetScalersToSample(
+            IScaleMonitorManager monitorManager,
+            ITargetScalerManager targetScalerManager,
+            IOptions<ScaleOptions> scaleOptions,
+            IConfiguration configuration)
+        {
+            var scaleMonitors = monitorManager.GetMonitors();
+            var targetScalers = targetScalerManager.GetTargetScalers();
+
+            var scaleMonitorsToSample = new List<IScaleMonitor>();
+            var targetScalersToSample = new List<ITargetScaler>();
+
+            // Check if TBS enabled on app level
+            if (scaleOptions.Value.IsTargetScalingEnabled)
+            {
+                HashSet<string> targetScalerFunctions = new HashSet<string>();
+                foreach (var scaler in targetScalers)
+                {
+                    string scalerUniqueId = GetTargetScalerFunctionUniqueId(scaler);
+                    if (!targetScalersInError.Contains(scalerUniqueId))
+                    {
+                        string assemblyName = GetAssemblyName(scaler.GetType());
+                        bool featureEnabled = configuration.GetSection(Constants.HostingConfigSectionName).GetValue<string>(assemblyName) == "1";
+                        if (featureEnabled)
+                        {
+                            targetScalersToSample.Add(scaler);
+                            targetScalerFunctions.Add(scalerUniqueId);
+                        }
+                    }
+                }
+
+                foreach (var monitor in scaleMonitors)
+                {
+                    string monitorUniqueId = GetScaleMonitorFunctionUniqueId(monitor);
+                    // Check if target based scaler exists for the function
+                    if (!targetScalerFunctions.Contains(monitorUniqueId))
+                    {
+                        scaleMonitorsToSample.Add(monitor);
+                    }
+                }
+            }
+            else
+            {
+                scaleMonitorsToSample.AddRange(scaleMonitors);
+            }
+
+            return (scaleMonitorsToSample, targetScalersToSample);
+        }
+
+        internal static ScaleVote GetAggregateScaleVote(IEnumerable<ScaleVote> votes, ScaleStatusContext context, ILogger logger)
+        {
+            ScaleVote vote = ScaleVote.None;
+            if (votes.Any())
```

Comment:
A logical simplification: just take the max scale vote of all the votes. I believe this logically equivalent to scale out if any workers are voting to scale out, and scale in iff all votes are scale in.

Reply:
This logic was migrated from RDS and it looks right to me as it also rely one worker count and if there is any votes. I do not see why we should change it.